Several nrfx samples fail to build

**Describe the bug**
After #64476 was merged, several nrfx samples (from the `hal_nordic` module) fail to build.

**To Reproduce**
Steps to reproduce the behavior:
1. `cd ../modules/hal/nordic/nrfx/samples/src/nrfx_saadc/simple_non_blocking`
2. `west build -b nrf52840dk_nrf52840`, see the following error:
```
error: NRFX_GPIOTE (defined at modules/hal_nordic/nrfx/Kconfig:65,
modules/hal_nordic/nrfx/Kconfig:65) is assigned in a configuration file, but is not directly user-
configurable (has no prompt). It gets its value indirectly from other symbols.
```
3. Even after changing `CONFIG_NRFX_GPIOTE=y` to `CONFIG_NRFX_GPIOTE0=y` in `prj.conf`, the sample still fails to build with quite a few errors like:
```
/home/andy/zephyrproject/modules/hal/nordic/nrfx/samples/src/nrfx_saadc/common/saadc_examples_common.c: In function 'gpiote_pin_toggle_task_setup':
/home/andy/zephyrproject/modules/hal/nordic/nrfx/samples/src/nrfx_saadc/common/saadc_examples_common.c:45:40: warning: passing argument 1 of 'nrfx_gpiote_channel_alloc' from incompatible pointer type [-Wincompatible-pointer-types]
   45 |     status = nrfx_gpiote_channel_alloc(&gpiote_channel);
      |                                        ^~~~~~~~~~~~~~~
      |                                        |
      |                                        uint8_t * {aka unsigned char *}
In file included from /home/andy/zephyrproject/modules/hal/nordic/nrfx/samples/src/nrfx_saadc/simple_non_blocking/../common/saadc_examples_common.h:37,
                 from /home/andy/zephyrproject/modules/hal/nordic/nrfx/samples/src/nrfx_saadc/common/saadc_examples_common.c:34:
/home/andy/zephyrproject/modules/hal/nordic/nrfx/drivers/include/nrfx_gpiote.h:249:60: note: expected 'const nrfx_gpiote_t *' but argument is of type 'uint8_t *' {aka 'unsigned char *'}
  249 | nrfx_err_t nrfx_gpiote_channel_alloc(nrfx_gpiote_t const * p_instance, uint8_t * p_channel);
      |                                      ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
```

**Expected behavior**
Samples provided in the `hal_nordic` module can be built successfuly.

**Impact**
Annoyance.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.4
 - Commit: 07c1e49f20a1b7cdd0b2b08ca10d9a90183cffc0

--- a/west.yml
+++ b/west.yml
@@ -183,7 +183,7 @@ manifest:
       groups:
         - hal
     - name: hal_nordic
-      revision: 4362d0fcdbac80942212bffc999654beac7ab30e
+      revision: dce8519f7da37b0a745237679fd3f88250b495ff
       path: modules/hal/nordic
       groups:
         - hal

